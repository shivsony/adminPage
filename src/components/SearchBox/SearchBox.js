import React from 'react';
import { makeStyles } from '@material-ui/core/styles';


const useStyles = makeStyles((theme) => ({
    root: {
      width: '100%', 
    },
    input: {
        width: '100%',
        padding: 10,
        fontFamily: "Roboto",
    }
}));

const SearchBox = ({placeholder, id, handleChange, value}) => {
    const classes = useStyles();
    return (
        <div className={classes.root}>
            <input className={classes.input} id={id} value={value} onChange={e=>handleChange(e.target.value)} placeholder={placeholder} />
        </div>
    )
}

export default SearchBox;