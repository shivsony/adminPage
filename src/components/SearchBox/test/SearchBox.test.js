
import ReactDOM from 'react-dom';
import renderer from 'react-test-renderer';


import SearchBox from '../SearchBox';


test('render without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<SearchBox id="user-search" placeholder="search" />, div);
});

test('matchs snapshot', () => {
    const tree = renderer.create(<SearchBox id="user-search" placeholder="search" value="shiv" onChange={()=>{}} />).toJSON();
    expect(tree).toMatchSnapshot();
});
