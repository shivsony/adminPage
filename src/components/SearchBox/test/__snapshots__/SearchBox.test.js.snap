// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matchs snapshot 1`] = `
<div
  className="makeStyles-root-1"
>
  <input
    className="makeStyles-input-2"
    id="user-search"
    onChange={[Function]}
    placeholder="search"
    value="shiv"
  />
</div>
`;
